tortoise/pydantic.py

```diff
@@ -0,0 +1,384 @@
+import functools
+import inspect
+import re
+import typing
+from typing import Any, Callable, Dict, List, Optional, Tuple, Type
+
+import tortoise
+from tortoise import fields, models
+
+if typing.TYPE_CHECKING:
+    from tortoise.models import Model
+
+
+# Pydantic is an optional dependency
+try:
+    import pydantic
+    from pydantic import BaseModel
+
+    @functools.lru_cache()
+    def _get_comments(cls: Type["Model"]) -> Dict[str, str]:
```

Comment:
This is actually a good idea, we should make this a standard.
Is it OK to populate the description from the comment if not explicitly defined?

What should be the order of preference?
1. `description=`
2. docstring
3. preceding comments

Note that a comment that starts with `#: ` sets the docstring, I think.

(Don't change this unless you REALLY want to, I'm thinking of moving this kind of thing up at a later time)

Reply:
I like this preceding comment. It is more natural and readable for me, because when I write my models I always create a description for it as a comment.  
Though the `description` parameter in my opinion is for the database fields primarily. This is why I made preceding comment support for Pydantic classes. For my API I need a better and more verbose description suport without writing the same thing more times. But I may not want the same description appear in the table field description. 
But it is only one (my) use case. For someone it maybe useful to write table comments this way.
Also note that this has multiline comment support, which is needed for API description. So you can use HTML, reStructuredText, MarkDown, or anything like that in here. The database descriptions are limited in size so don't think that they are created for this kind of documentation.

As far as I know `#:` is only in Epydoc which is not developed since 2008. Others don't have variable level  docstrings.
It is not hard to rewrite to use only comments started with `#:`if you prefer that. Just it can be a collision with reStructuredText format (which PyCharm uses by default), when you start a comment like eg. `#:return: ...`.

Reply:
Sphinx autodoc (which we use) uses `#:`, but you are right, it is a tooling convention, the Python parser doesn't care.

I see, we really need two levels of description. a short one, and a long one.
The long one can default to the short one if provided.

I think I'll introduce a `long_description` var, that either is a param, or it tries to get the preceding comments like this.

Then the sometimes HUGE descriptions generated by the enum fields could go into long_description by default?

Reply:
I created that automatic enum description because when you see enums in the DB, you see only numbers. So for me it is a help to be able to know which number means what. So that should be included in the DB (short_description) IMO. 

Anyway I like the idea. And that way the description can be from `describe_model` too.

Reply:
I looked up comment limits, and MySQL has a limit of 255 for a column (on v5.5). It is upped to 1023 since v5.6 (which is also now quite old).
We have to just be careful to limit the comment length for the generated comments.